metricbeat/module/elasticsearch/index_recovery/data_xpack.go

```diff
@@ -38,23 +41,23 @@ func eventsMappingXPack(r mb.ReporterV2, m *MetricSet, content []byte) error {
 	for indexName, indexData := range data {
 		indexData, ok := indexData.(map[string]interface{})
 		if !ok {
-			continue
+			return fmt.Errorf("%v is not a map", indexName)
```

Comment:
My thinking with the new approach is:
1. It's unlikely that one is wrong and others are not, and also
2. In the case when some data might be erroneous, is it better to make a best effort or is it better to fail fast? I decided to go with the latter because I think it makes it easier to spot an issue.

Reply:
Let's move forward with the way it is at the moment and fix it if it breaks.